central/authprovider/service/service_impl.go

```diff
@@ -319,12 +320,13 @@ func (s *serviceImpl) ExchangeToken(ctx context.Context, request *v1.ExchangeTok
 		Test:        testMode,
 	}
 
+	userMetadata, err := authproviders.CreateRoleBasedIdentity(sac.WithAllAccess(ctx), provider, authResponse)
+	if err != nil {
+		log.Warnf("error creating role based identity: %v", err)
+	}
+	userPkg.LogSuccessfulUserLogin(log, userMetadata)
+
```

Comment:
I'd say we only want to log successful user logins _iff_ we are not in test mode. I'm not sure how helpful it would be to log test mode users.

Reply:
From API perspective, the test mode is just a flag. This flag does not prevent API consumers from obtaining the resulting token - so unauthorised access is still possible even when test mode is enabled.